doc/guides/west/repo-tool.rst

```diff
@@ -257,6 +257,28 @@ If you would like ``west update`` to keep local branches checked out as
 long as they point to commits that are descendants of the new
 ``manifest-rev``, use the ``-k`` (``--keep-descendants``) option.
 
+.. note::
+
+   ``west update --rebase`` will fail in projects that have git conflicts
+   between your branch and new commits brought in by the manifest. You
+   should immediately resolve these conflicts as you usually do with
+   ``git``, or you can use ``git -C <project_path> rebase --abort`` to
+   ignore incoming changes for the moment.
+
+   With a clean working tree, a plain ``west update`` never fails
+   because it does not trying to hold on to your commits and simply
```

Comment:
```suggestion
   because it does not try to hold on to your commits and simply
```

Reply:
Good catch, thanks for the thorough review.

I'm going to click on "commit suggestion". Last time I was in a discussion about this (a very long time ago) this project didn't allow github's most popular "squash commits" feature, it was "force pushes only (#14444) However you're the second person suggesting a fixup commit like this in just 2 days so it looks like the policy is not as strict now? Plus this is a very small commit so very little confusion in the worst of cases.

Reply:
Policy hasn't changed; you can use the commit suggestion button, but you need to fetch the resulting fix-up commits, squash, and force push.

Reply:
Done.

> but you need to fetch the resulting fix-up commits, squash, and force push.

Thanks. For changing one word it's not worth it.

Reply:
> Thanks. For changing one word it's not worth it.

Your call, of course; it nevertheless has value as a convenient way to provide the exact changes that are being requested.

Reply:
> > but you need to fetch the resulting fix-up commits, squash, and force push.

> Thanks. For changing one word it's not worth it.

Agree, for a single word change like this, not so useful to hit `commit` button, but syntax highlighting the suggested change is very nice.

And if there are 3+ commit suggestions like this, it's much easier to just commit them all, and then `git fetch` + `git rebase` with squash is easy.